### ISSUE TITLE
React componentDidCatch not able to handle exceptions in Server Side Rendering

### ISSUE THREAD

**Issue by ian-craig** [reactjs/React.NET#503]
When using componentDidCatch and the ErrorBoundary pattern, I would expect to be able to handle JS exceptions in JS. However, these seem to bubble up to React.Net and throw in C# no matter what I do.

"Error while rendering "CommentsBox" to "react_NcDXGhXrkmj56hcuuqRug": TypeError: Unable to get property 'one' of undefined or null reference at render (Script Document:257:13) at resolve (React.Core.Resources.react.generated.js:4273:6) at render (React.Core.Resources.react.generated.js:4373:8) at read (React.Core.Resources.react.generated.js:4349:8) at renderToString (React.Core.Resources.react.generated.js:4607:4) at Global code (Script Document [3]:1:1) Line: 257 Column:13"

Using the exact sample from [React.NET/src/React.Sample.CoreMvc/wwwroot/js/Sample.jsx](https://github.com/reactjs/React.NET/blob/f408fa4bcada504618ccdda8c61c61e2dbb7ace6/src/React.Sample.CoreMvc/wwwroot/js/Sample.jsx)

With the following in my .cshtml
```cs
@React.Web.Mvc.HtmlHelperExtensions.React(Html, "CommentsBox", new { throwRenderError = true });
```

When running the same code in a browser the ErrorBoundary catches exceptions correctly.

Is there a recommended way of handling exceptions that I am missing?

**Comment by dustinsoftware** [comment]
Thanks for asking! Support for this has been implemented and should be
available in the next release. You can use the built DLLs from the build
server if you need this behavior today.

**Comment by dustinsoftware** [comment]
Released in 3.3. Sample use is [here](https://github.com/reactjs/React.NET/blob/f408fa4bcada504618ccdda8c61c61e2dbb7ace6/src/React.Sample.CoreMvc/wwwroot/js/Sample.jsx#L123)
